src/line_cursor.ts

```diff
@@ -410,10 +448,14 @@ export class LineCursor extends Marker {
    * if so, update the cursor location (and any highlighting) to
    * match.
    *
-   * This works reasonably well but has some glitches, most notably
-   * that if the cursor is not on a block (e.g. it is on a connection
-   * or the workspace) then it will remain visible in its previous
-   * location until a cursor key is pressed.
+   * Doing this only when getCurNode would naturally be called works
+   * reasonably well but has some glitches, most notably that if the
+   * cursor was not on a block (e.g. it was on a connection or the
+   * workspace) when the user selected a block then it will remain
+   * visible in its previous location until some keyboard navigation occurs.
+   *
+   * To ameliorate this, the LineCursor constructor adds an event
+   * listener that calls getCurNode in response to SELECTED events.
    *
    * TODO(#97): Remove this hack once Blockly is modified to update
```

Comment:
Delete "TODO(#97)". This fixes the issue stated in 97 and we'll rewrite much more than just this function when modifying core Blockly for this.